test/main/chains/gas/index.test.js

```diff
@@ -0,0 +1,227 @@
+import { intToHex } from '@ethereumjs/util'
+import { createGasCalculator } from '../../../../main/chains/gas'
+import { gweiToHex } from '../../../util'
+
+describe('#gasCalculator', () => {
+  let gasCalculator
+  let feeHistory
+
+  const calculateGas = async () => gasCalculator.calculateGas(feeHistory)
+  const calculateMaxPriority = async () => {
+    const { maxPriorityFeePerGas } = await calculateGas()
+    return maxPriorityFeePerGas
+  }
+
+  beforeEach(() => {
+    gasCalculator = createGasCalculator()
+    feeHistory = []
+  })
+
+  it('calculates the base fee for the next couple blocks', async () => {
+    feeHistory = [
+      { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+      { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+      { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+      { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+      { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+      { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+      { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+      { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+      { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+      { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+      { baseFee: 182, rewards: [] }
+    ]
+    const { maxBaseFeePerGas } = await calculateGas()
+    expect(maxBaseFeePerGas).toBe('0xe7')
+  })
+
+  describe('calculating priority fee', () => {
+    it('calculates the priority fee for the next block based on normal blocks', async () => {
+      // all blocks with gas ratios between 0.1 and 0.9 will be considered for calculating the median priority fee
+      feeHistory = [
+        { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+        { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+        { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+        { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+        { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+        { baseFee: 8, gasUsedRatio: 0, rewards: [0] },
+        { baseFee: 8, gasUsedRatio: 0.1801134637893198, rewards: [1000000000] },
+        { baseFee: 8, gasUsedRatio: 0.23114498292513627, rewards: [2000000000] },
+        { baseFee: 8, gasUsedRatio: 0.17942918604838942, rewards: [1000000000] },
+        { baseFee: 8, gasUsedRatio: 0.12024061496050893, rewards: [4000000000] },
+        { baseFee: 182, rewards: [] }
+      ]
+      expect(await calculateMaxPriority()).toBe('0x77359400')
```

Comment:
dont need to await here, can use the slightly more idiomatic `return expect(calculateMaxPriority()).resolves.toBe('0x77359400')`